falcon.go

```diff
@@ -193,7 +189,7 @@ func (sig *CTSignature) SaltVersion() byte {
 //
 // Returns an error if pubkey is invalid.
 func (pub *PublicKey) Coefficients() (h [N]uint16, err error) {
-	r := C.falcon_det1024_pubkey_coeffs((*C.uint16_t)(&h[0]), unsafe.Pointer(pub))
+	r := C.falcon_det1024_pubkey_coeffs((*C.uint16_t)(&h[0]), unsafe.Pointer(&(*pub)))
```

Comment:
what is a point of defererensing and then taking an address again?

Reply:
it is a common way to access slices and array on cgo.
we use this on curve25519.go
and it is also mention here
https://pkg.go.dev/unsafe#Pointer

Reply:
I'm asking about `pub` that is not an array

Reply:
`pub` is defined as instance of type `PublicKey`which is defined as
`type PublicKey [PublicKeySize]byte`